### PR TITLE
refactor(compiler): Update %COMP% string to %C% and updated regex to not search globally.

### PR DESCRIPTION
**Pull request by ngwattcos**
Changed the %COMP% component placeholder in the temporary attribute to %C% to shorten the attribute.

Changed COMPONENT_REGEX in dom_renderer.ts to not search the string globally, since %C% will only appear once in an attribute selector.

Shortening the attribute allows a smaller bundle size and a smaller download before the platform-server shis the component attribute with component ID.

All spec tests involving %COMP% have been updated to use %C%.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ x ] Other... Please describe:
The API is slightly changed because the outputted %COMP% string in angular attributes is now %C%. Tests have been updated and re-run to verify this works.

## What is the current behavior?

The component placeholder variable in the angular attribute currently outputs %COMP%.

Issue Number: N/A

## What is the new behavior?
outputted %COMP% string is replaced with %C%.

## Does this PR introduce a breaking change?

- [ ] Yes
- [ x ] No

### CODE REVIEW

File: packages/compiler/src/style_compiler.ts [hunk 1]
```diff
@@ -6,90 +6,135 @@
...
+  identifierName,
+} from "./compile_metadata";
+import { ViewEncapsulation } from "./core";
+import * as o from "./output/output_ast";
+import { ShadowCss } from "./shadow_css";
+import { UrlResolver } from "./url_resolver";
+import { OutputContext } from "./util";
 
-const COMPONENT_VARIABLE = '%COMP%';
+const COMPONENT_VARIABLE = "%C%";
```
Comment (kseamon) [inline packages/compiler/src/style_compiler.ts:22]: Use single-quote '

(more or less, try to follow the google style guide or at least just match nearby code)
https://google.github.io/styleguide/jsguide.html

File: packages/platform-browser/src/dom/dom_renderer.ts [hunk 2]
```diff
@@ -6,23 +6,32 @@
...
+  svg: "http://www.w3.org/2000/svg",
+  xhtml: "http://www.w3.org/1999/xhtml",
+  xlink: "http://www.w3.org/1999/xlink",
+  xml: "http://www.w3.org/XML/1998/namespace",
+  xmlns: "http://www.w3.org/2000/xmlns/",
 };
 
-const COMPONENT_REGEX = /%COMP%/g;
-const NG_DEV_MODE = typeof ngDevMode === 'undefined' || !!ngDevMode;
+const COMPONENT_REGEX = /%C%/;
```
Comment (kseamon) [inline packages/platform-browser/src/dom/dom_renderer.ts:31]: As I noted in chat, please verify the runtime performance impact of removing /g as well as seeing what difference $ would make (probably with jsperf)

Comment (JoostK) [reply packages/platform-browser/src/dom/dom_renderer.ts:31]: I don't see why the `g` flag can be dropped. For a selector such as `one two`, a shimmed selector `one[_ngcontent-%COMP%] two[_ngcontent-%COMP%]` is generated which requires both occurrences of `%COMP%` to be replaced.

For the `HOST_ATTR` and `CONTENT_ATTR` case, we _can_ however update `shimHostAttribute` and `shimContentAttribute` to simply append the component id to the `_nghost-` and `_ngcontent-` strings.

Comment (ngwattcos) [reply packages/platform-browser/src/dom/dom_renderer.ts:31]: Right, I misunderstood the regex expression. I'll also update to simply use the component id.

File: packages/compiler/src/style_compiler.ts (hunk 1 above)
Comment (ngwattcos) [reply packages/compiler/src/style_compiler.ts:22]: Yikes, every time I try to save, it auto-formats to a single quote.

**Pull request closed without merging**